src/coreComponents/common/format/table/TableLayout.hpp

```diff
@@ -51,68 +55,328 @@ class TableLayout
+    size_t m_cellWidth = 0;
+
+    /**
+     * @brief Default constructor.
+     * Initializes a column with default values.
+     */
+    Column();
+
+    /**
+     * @brief Constructor to initialize a column with a specific `CellLayout`.
+     * @param cell The `CellLayout` object to initialize the column.
+     *
+     */
+    Column( TableLayout::CellLayout cellLayout );
+
+    /**
+     * @brief Sets the name of the column.
+     * @param name The name to set for the column.
+     * @return The current column object.
+     */
+    Column & setName( string_view name );
+
+    /**
+     * @brief Hides the column.
+     * @return The current column objec.
+     */
+    Column & hide();
+
+    /**
+     * @brief Sets the maximum string size for the column.
+     * @param size The size to set as the maximum string length.
+     */
+    void setMaxStringSize( size_t const size );
+
+    /**
+     * @brief Gets the maximum string size of the column.
+     * @return size_t The maximum string size of the column.
+     */
+    size_t getMaxStringSize() const;
+
+    /**
+     * @brief Adds multiple sub-columns to the column.
+     * @param subColName A list of sub-column names to add.
+     * @return The current column object
+     */
+    TableLayout::Column & addSubColumns( std::initializer_list< TableLayout::Column > subCol );
+
+    /**
+     * @brief Adds multiple sub-columns to the column.
+     * @param subColName A list of sub-column names to add.
+     * @return The current column object
+     */
+    TableLayout::Column & addSubColumns( std::initializer_list< string > subColName );
+
+    /**
+     * @brief Adds a single sub-column to the column.
+     * @param subColName The name of the sub-column to add.
+     * @return The current column object.
+     */
+    TableLayout::Column & addSubColumns( string const & subColName );
+
+    /**
+     * @brief Sets the header alignment for the column.
+     * @param headerAlignment The alignment to set for the column header (left, right, or center).
+     * @return The current column object
+     */
+    TableLayout::Column & setHeaderAlignment( Alignment headerAlignment );
+
+    /**
+     * @brief Sets the values alignment for the column.
+     * @param headerAlignment The alignment to set for the column values (left, right, or center).
+     * @return The current column object
+     */
+    TableLayout::Column & setValuesAlignment( Alignment valueAlignment );
+
+    /**
+     * @brief Checks if the column has any child columns.
+     * @return bool True if the column has child columns, otherwise false.
+     */
+    bool hasChild() const;
+
+    /**
+     * @brief Checks if the column has a parent column.
+     * @return bool True if the column has a parent, otherwise false.
+     */
+    bool hasParent() const;
+
+private:
+    /// The maximum string size of the column.
+    size_t m_maxStringSize;
+  };
+
+  /**_est columns / sub columns.
```

Comment:
_est?

Reply:
```suggestion
  /**
   * Iterator to loop over all columns, starting by the deepest sub columns, then to their parents, then to their siblings.
```
(and please take the code style max line width in consideration)